backend/functions/lib/adapters/gateways/consolidations.cosmosdb.mongo.repository.test.ts

```diff
@@ -0,0 +1,154 @@
+import { ApplicationContext } from '../types/basic';
+import { createMockApplicationContext } from '../../testing/testing-utilities';
+import MockData from '../../../../../common/src/cams/test-utilities/mock-data';
+import ConsolidationOrdersCosmosMongoDbRepository from './consolidations.cosmosdb.mongo.repository';
+import { MongoCollectionAdapter } from './mongo/mongo-adapter';
+import QueryBuilder from '../../query/query-builder';
+import { closeDeferred } from '../../defer-close';
+import { getCamsError } from '../../common-errors/error-utilities';
+
+describe('Consolidations Repository tests', () => {
+  let context: ApplicationContext;
+  let repo: ConsolidationOrdersCosmosMongoDbRepository;
+  const { and, contains, equals, orderBy } = QueryBuilder;
+
+  beforeEach(async () => {
+    context = await createMockApplicationContext();
+    repo = new ConsolidationOrdersCosmosMongoDbRepository(context);
+  });
+
+  afterEach(async () => {
+    await closeDeferred(context);
+    jest.restoreAllMocks();
+  });
+
+  test('should search on consolidations by court division code or consolidationId', async () => {
+    const consolidationId = '823688b3-9e0f-4a02-a7cb-89380e6ad19e';
+    const consolidationOrder = MockData.getConsolidationOrder({
+      override: { consolidationId, courtDivisionCode: '081' },
+    });
+    const findSpy = jest
+      .spyOn(MongoCollectionAdapter.prototype, 'find')
+      .mockResolvedValue([consolidationOrder]);
+    const query = QueryBuilder.build(
+      and(
+        contains<string[]>('courtDivisionCode', [consolidationOrder.courtDivisionCode]),
+        equals<string>('consolidationId', consolidationOrder.consolidationId),
+      ),
+    );
+    const results = await repo.search({
+      divisionCodes: ['081'],
+      consolidationId: consolidationOrder.consolidationId,
+    });
+
+    expect(results).toEqual([consolidationOrder]);
+    expect(results.length).toEqual(1);
+    expect(findSpy).toHaveBeenCalledWith(query, orderBy(['orderDate', 'ASCENDING']));
+  });
+
+  test('should search on consolidations with an empty query', async () => {
+    const consolidationOrders = MockData.buildArray(MockData.getConsolidationOrder, 5);
+    const findSpy = jest
+      .spyOn(MongoCollectionAdapter.prototype, 'find')
+      .mockResolvedValue(consolidationOrders);
+    const results = await repo.search();
+
+    expect(results).toEqual(consolidationOrders);
+    expect(findSpy).toHaveBeenCalledWith(null, orderBy(['orderDate', 'ASCENDING']));
+  });
+
+  test('should call delete on a consolidation order', async () => {
+    const consolidationId = '823688b3-9e0f-4a02-a7cb-89380e6ad19e';
```

Comment:
Does this id mean something?

Reply:
I think this was hanging around from when we were doing tests against the actual database